src/Dispatcher/ZmqDispatcher.php

```diff
@@ -26,181 +29,119 @@ class ZmqDispatcher implements DispatcherInterface
     /**
      * @var string
      */
-    protected $application;    
+    protected $application;
     
     /**
      * @var string
      */
     protected $environment;
-    
-    /**
-     * @var array
-     */
-    protected $fieldsToFilter;
-    
-    /**
-     * Indexes to be filtered from requests are replaced with this string.
-     * @var string
-     */
-    protected $filterMask;
-    
+
     /**
      * @var boolean
      */
     protected $isConnected;
 
     /**
+     * @var array
+     */
+    protected $exceptions = [];
+
+    /**
      * ZmqDispatcher constructor.
+     * @param ZMQSocket $socket
      * @param array $brokers ZeroMQ Broker addresses (tcp://host:port).
      * @param $application
      * @param string $environment String that identifies the environment the app is in.
-     * @param array $fieldsToFilter Array of identifiers in post or get data that should be filtered from the log data.
-     * @param string $filterMask String to replace the filtered data with.
      */
-    public function __construct(array $brokers, $application, $environment, array $fieldsToFilter = array(), $filterMask = '*****')
+    public function __construct(ZMQSocket $socket, array $brokers, $application, $environment)
```

Comment:
Please replace the ZMQSocket dependency with a dependency to a factory that creates them. My Idea was that the application should not implement any exception handling if it wants to use this dispatcher